modules/logging/filters.go

```diff
@@ -65,6 +68,35 @@ func (DeleteFilter) Filter(in zapcore.Field) zapcore.Field {
 	return in
 }
 
+// hash returns the first 4 bytes of the SHA-256 hash of the given data as hexadecimal
+func hash(s string) string {
+	return fmt.Sprintf("%.4x", sha256.Sum256([]byte(s)))
```

Comment:
How did you decide on 4 bytes? It's shorter than what I would think it takes for inevitable collisions not to become a frequent occurrence – but I don't have data to recommend an alternative minimum.
Anecdotal: [nginx uses 16 bytes for request ids](https://nginx.org/en/docs/http/ngx_http_core_module.html#var_request_id) (but those are not based on input, of course).

Reply:
See @francislavoie's comment: https://github.com/caddyserver/caddy/pull/4434#issuecomment-978012125 (I indeed copied Git's behavior).

Also, I don't think that we care much about collisions here as the main use case is redacting data but being able to detect different patterns when analyzing the logs (being 100% reliable is not really useful for this use case). Also, a short hash allows reducing the size of the logs on disk.